File: leetcode-submissions/0557-reverse-words-in-a-string-iii/solution.java
class Solution {
    public String reverseWords(String s) {
        //charArray and two pointer
        String[] sArr = s.split("\\s+");
        StringBuilder sb = new StringBuilder();
        for(int i = 0; i < sArr.length; i++) {
            char[] curr = sArr[i].toCharArray();
            int left = 0;
            int right = curr.length - 1;
            while(left < right) {
                char temp = curr[left];
                curr[left] = curr[right];
                curr[right] = temp;
                left++;
                right--;
            }
            sArr[i] = String.valueOf(curr);

            if (i < sArr.length - 1) {
                sb.append(sArr[i]);
                sb.append(" ");
            }
        }
        sb.append(sArr[sArr.length - 1]);
        return sb.toString();   
    }
}

File: leetcode-submissions/0557-reverse-words-in-a-string-iii/solution.py
class Solution:
    def reverseWords(self, s: str) -> str:
        word_list = []

        for word in s.split(" "):
            reverse_list = list(word)
            
            left = 0
            right = len(reverse_list) - 1

            while left < right:
                helper = reverse_list[left]
                reverse_list[left] = reverse_list[right]
                reverse_list[right] = helper
                left += 1
                right -= 1

            word_list.append("".join(reverse_list))
            word_list.append(" ")
        
        
        return "".join(word_list[0:-1])
        
